src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -40,29 +41,41 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
                 return new HttpResponseMessage(HttpStatusCode.NotFound);
             }
 
-            // Authorize the request
+            // Determine the authorization level of the request
             SecretManager secretManager = (SecretManager)controllerContext.Configuration.DependencyResolver.GetService(typeof(SecretManager));
+            AuthorizationLevel authorizationLevel = AuthorizationLevelAttribute.GetAuthorizationLevel(request, secretManager, functionName: function.Name);
+
+            // Dispatch the request
             HttpTriggerBindingMetadata httpFunctionMetadata = (HttpTriggerBindingMetadata)function.Metadata.InputBindings.FirstOrDefault(p => p.Type == BindingType.HttpTrigger);
             bool isWebHook = !string.IsNullOrEmpty(httpFunctionMetadata.WebHookType);
-            if (!isWebHook && !AuthorizationLevelAttribute.IsAuthorized(request, httpFunctionMetadata.AuthLevel, secretManager, functionName: function.Name))
-            {
-                return new HttpResponseMessage(HttpStatusCode.Unauthorized);
-            }
-
             HttpResponseMessage response = null;
             if (isWebHook)
             {
-                // This is a WebHook request so define a delegate for the user function.
-                // The WebHook Receiver pipeline will first validate the request fully
-                // then invoke this callback.
-                Func<HttpRequestMessage, Task<HttpResponseMessage>> invokeFunction = async (req) =>
+                if (authorizationLevel == AuthorizationLevel.Admin)
```

Comment:
This is the main change - here for Admin requests we dispatch directly to the user method